dfusion_rust_core/src/models/solution.rs

```diff
@@ -3,24 +3,30 @@ use super::*;
 use log::info;
 
 use byteorder::{BigEndian, ByteOrder};
+use std::collections::HashMap;
 use std::iter::once;
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct Solution {
-    pub prices: Vec<u128>,
+    // token_id => price
+    pub prices: HashMap<u16, u128>,
     pub executed_buy_amounts: Vec<u128>,
     pub executed_sell_amounts: Vec<u128>,
 }
 
 impl Solution {
     pub fn trivial(num_orders: usize) -> Self {
         Solution {
-            prices: vec![],
+            prices: HashMap::new(),
             executed_buy_amounts: vec![0; num_orders],
             executed_sell_amounts: vec![0; num_orders],
         }
     }
 
+    pub fn max_token(&self) -> Option<u16> {
+        self.prices.keys().max().copied()
+    }
+
```

Comment:
I would add another method to `Solution` for getting the price for a token ID `pub fn price(token_id: u16) -> u128 { /*...*/ }` as I see that it is used a lot.

Reply:
Yeah, we could probably get rid of the `get_token_price` method on the SnappSolution.

Reply:
Will do! What to do with all the nice doc strings?

Reply:
You still need the `snapp_objective_value` method I think. You can move the `get_token_price` doc string for `Solution::price` :D.